pkg/agent/cluster/controllers.go

```diff
@@ -0,0 +1,42 @@
+package cluster
+
+import (
+	"context"
+
+	clusterController "github.com/rancher/rancher/pkg/controllers/user"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	"k8s.io/client-go/rest"
+)
+
+func RunControllers() error {
+	logrus.Info("Starting user controllers")
+	c, err := rest.InClusterConfig()
+	if err != nil {
+		return err
+	}
+
+	scaledContext, err := config.NewScaledContext(*c)
+	if err != nil {
+		return err
+	}
+
+	userContext, err := config.NewUserContext(scaledContext, *c, "")
```

Comment:
If "ClusterName" on UserOnlyContext is never used, can we drop the field, otherwise we should properly populate the value.